DSCResources/MSFT_xHostsFile/MSFT_xHostsFile.psm1

```diff
@@ -0,0 +1,267 @@
+$Script:hostsFilePath = "${env:windir}\system32\drivers\etc\hosts"
+
+Import-LocalizedData -BindingVariable LocalizedData -filename MSFT_xHostsFile.psd1 -BaseDirectory $PSScriptRoot -Verbose
+
+function Get-TargetResource
+{
+    [OutputType([Hashtable])]
+    param (
+        [parameter(Mandatory = $true)]
+        [string]
+        $HostName,
+        [parameter(Mandatory = $true)]
+        [string]
+        $IPAddress
+    )
+
+    $configuration = @{
+        HostName = $HostName
+        IPAddress = $IPAddress
+    }
+
+    Write-Verbose $localizedData.checkingHostsFileEntry
+    try {
+        if (Test-HostEntry -IPAddress $IPAddress -HostName $HostName) {
+            Write-Verbose ($localizedData.hostsFileEntryFound -f $HostName, $IPAddress)
+            $configuration.Add('Ensure','Present')
+        } else {
+            Write-Verbose ($localizedData.hostsFileEntryNotFound -f $HostName, $IPAddress)
+            $configuration.Add('Ensure','Absent')
+        }
+        return $configuration
+    } catch {
+        $exception = $_
+        Write-Verbose ($LocalizedData.anErrorOccurred -f $name, $exception.message)
+        while ($null -ne $exception.InnerException)
+        {
+            $exception = $exception.innerException
+            Write-Verbose ($LocalizedData.innerException -f $name, $exception.message)
+        }
+    }
+}
+
+function Set-TargetResource
+{
+    param (
+        [parameter(Mandatory = $true)]
+        [string]
+        $HostName,
+        [parameter(Mandatory = $true)]
+        [string]
+        $IPAddress,
+        [parameter()]
+        [ValidateSet('Present','Absent')]
+        [string]
+        $Ensure = 'Present'
+    )
+
+    try {
+        if ($Ensure -eq 'Present') {
+            Write-Verbose ($localizedData.creatingHostsFileEntry -f $HostName, $IPAddress)
+            Add-HostEntry -IPAddress $IPAddress -HostName $HostName
+            Write-Verbose ($localizedData.hostsFileEntryAdded -f $HostName, $IPAddress)
+        } else {
+            Write-Verbose ($localizedData.removingHostsFileEntry -f $HostName, $IPAddress)
+            Remove-HostEntry -IPAddress $IPAddress -HostName $HostName
+            Write-Verbose ($localizedData.hostsFileEntryRemoved -f $HostName, $IPAddress)
+        }
+    } catch {
+        $exception = $_
+        Write-Verbose ($LocalizedData.anErrorOccurred -f $name, $exception.message)
+        while ($null -ne $exception.innerException) {
+            $exception = $exception.innerException
+            Write-Verbose ($LocalizedData.innerException -f $name, $exception.message)
```

Comment:
Why are we writing verbose here instead of throwing or writing an error message?